### ISSUE TITLE
Filename of `js.js` causes watched tests to re-run endlessly

### ISSUE THREAD

**Issue by brandonweiss** [avajs/ava#1747]
## Description

Thanks for AVA! ❤️

It would appear that if I name a file `js.js` and run `ava` with the `--watch` flag it triggers a re-run endlessly.

I know, it's a bit of an unusual name for a file. But for a variety of reasons the files are named after other files’ extensions, e.g. `css.js`, `html.js`, `jsx.js`, and thus `js.js`.

I imagine the name must be tripping up a glob or a regex or something.

### Command-Line Arguments

```
ava --verbose --serial --watch
```

## Environment

```
Node.js v9.8.0
darwin 17.4.0
ava 0.25.0
npm 5.6.0
```

**Comment by brandonweiss** [comment]
Huh, OK now I'm not so sure. It seemed like I was very reliably able to stop the re-running by naming the file `javascript.js` and then start it by naming it `js.js`, but I've moved onto another feature and even though the file is name `javascript.js` I'm still encountering the endless re-running. 🤔

**Comment by novemberborn** [comment]
@brandonweiss have a look at https://github.com/avajs/ava/blob/master/docs/recipes/watch-mode.md#debugging — it should give you some clues as to what file system activity is causing the tests to rerun.

(I'm closing this issue for housekeeping purposes, but let's keep the conversation going.)

**Comment by brandonweiss** [comment]
Ah, thanks for the debugging tip!

OK, so the filename being `js.js` was a total red herring. Sorry about that! What was happening is my package needs to create files on the filesystem and then the tests assert the files are there or have changed in some way. I was writing those files to `tmp/tests`, and it wasn't a problem because the files were all `.css`, `.html`, or `.html.jsx` files. But when I started testing `.js` files create a `.js` file in the test trigger another re-run. 🤦🏼‍♂️

So the solution is I just needed to exclude `tmp` from the source files being watched in my `package.json` like so:

```json
{
  "ava": {
    "source": [
      "!tmp/**/*"
    ]
  }
}
```

Again, sorry about that! Thanks for your help!
